docs/setup-dev-environment.md

```diff
@@ -1,43 +1,116 @@
 # Setting up a local development environment
 
-This document is intended to guide you through setting up a local development environment for
-Colouring London. This guide assumes you already have Ubuntu 20.04 server installed, typically
-installed in a virtual environment such a Virtual Box and are able to SSH into your Ubuntu
-installation for convenience.
+This document is intended to guide you through setting up a local development environment for the Colouring London application. This guide assumes you already have either already have access to an machine with Ubuntu 18.04 or 20.04 installed, or can use VirtualBox to set up an Ubuntu virtual machine as below:
+
+<details>
+<summary>
+Configuring an Ubuntu VM in VirtualBox
+</summary><p></p>
+
+Here we explain how to use VirtualBox and SSH into your Ubuntu installation for convenience.
+
+When setting up the VirtualBox VM, consider the size of the database you intend to load for use with the application. Consult the [:house: Loading the building data](#house-loading-the-building-data) section of this guide and decide whether you will be using a full city database or will load test data from OSM.
+
+For "Colouring London", we have found that the size of the database means that a VM with access to 50GB of storage is appropriate. If you are using the OSM test data, the default storage settings in VirtualBox should suffice.
+
+##### In either case, you should set the memory to `2048` MB.
+
+If you a running Ubuntu in a virtual environment you will need to configure networking to forward ports from the guest to the host. For Virtual Box the following was configured under NAT port forwarding (found under `Settings -> Network -> Advanced -> Port Forwarding`).
+
+Name     | Protocol  | Host Port  | Guest Port
+-------- | --------- | ---------- | -----------
+app      | TCP       | 8080       | 3000
+app_dev  | TCP       | 3001       | 3001
+ssh      | TCP       | 4022       | 22
+
+The `app_dev` mapping is used in development by Razzle which rebuilds and serves client side assets on the fly.
+
+To run the commands in the rest of this setup guide, either `ssh` into the VirtualBox environment or open the terminal within the Ubuntu GUI.
+
+If you wish to `ssh`, you will first need to open the terminal in Ubuntu and run the following.
```

Comment:
```suggestion
If you wish to `ssh`, you will first need to open the terminal in Ubuntu and run the following:
```

very minor stylistic change (proposed just here for now to check is it welcomed)

Reply:
ok yes, I can change throughout to make consistent